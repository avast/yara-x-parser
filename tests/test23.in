//Rule comment
rule test
{
	//Rule block comment
	strings:
		$b = "bar"
	condition:
		$b and not defined $b
}
