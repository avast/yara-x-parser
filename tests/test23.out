SOURCE_FILE@0..110
  RULE@0..109
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..109
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..71
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$b"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"bar\""
      WHITESPACE@71..73 "\n\t"
      CONDITION@73..107
        CONDITION_KW@73..82 "condition"
        COLON@82..83 ":"
        WHITESPACE@83..86 "\n\t\t"
        BOOLEAN_EXPR@86..107
          BOOLEAN_TERM@86..88
            VARIABLE@86..88 "$b"
          WHITESPACE@88..89 " "
          AND_KW@89..92 "and"
          WHITESPACE@92..93 " "
          BOOLEAN_EXPR@93..107
            BOOLEAN_TERM@93..107
              NOT_KW@93..96 "not"
              WHITESPACE@96..97 " "
              BOOLEAN_TERM@97..107
                DEFINED_KW@97..104 "defined"
                WHITESPACE@104..105 " "
                BOOLEAN_TERM@105..107
                  VARIABLE@105..107 "$b"
      WHITESPACE@107..108 "\n"
      R_BRACE@108..109 "}"
  WHITESPACE@109..110 "\n"
